prow/cmd/phaino/local.go

```diff
@@ -246,7 +250,87 @@ func convertToLocal(ctx context.Context, log *logrus.Entry, pj prowapi.ProwJob,
+			defaultGcloudConfigPath := path.Join(os.Getenv("HOME"), ".config/gcloud")
+			if _, err := os.Stat(defaultGcloudConfigPath); !os.IsNotExist(err) {
+				localArgs = append(localArgs, "-v", defaultGcloudConfigPath+":"+gcloudDefaultConfigMountPath)
+				// Overwrite the gcloud config path, as per https://stackoverflow.com/a/48343135
+				localArgs = append(localArgs, "-e", "CLOUDSDK_CONFIG="+gcloudDefaultConfigMountPath)
+			} else {
+				logrus.Warning("The default gcloud credentials does not exist on your local machine, thus gloud authentication won't work in the container")
+			}
+		}
+	}
+
+	// Setup kubeconfig
+	if opt.useLocalKubeconfig || envExists(container.Env, kubeconfigEnvKey) {
+		kubeconfigEnvVarVal := os.Getenv(kubeconfigEnvKey)
+		// If KUBECONFIG is not empty, also mount the kubeconfig files to the
+		// container
+		if kubeconfigEnvVarVal != "" {
+			localArgs = append(localArgs, "-e", kubeconfigEnvKey+"="+kubeconfigEnvVarVal)
+			inexistentKubeconfigFiles := make([]string, 0)
+			for _, f := range strings.Split(kubeconfigEnvVarVal, string(os.PathListSeparator)) {
+				if _, err := os.Stat(f); !os.IsNotExist(err) {
+					localArgs = append(localArgs, "-v", f+":"+f+":ro")
+				} else {
+					inexistentKubeconfigFiles = append(inexistentKubeconfigFiles, f)
+				}
+			}
+			if len(inexistentKubeconfigFiles) == 0 {
+				localArgs = append(localArgs, "-e", kubeconfigEnvKey+"="+kubeconfigEnvVarVal)
+			} else {
+				logrus.Warningf("kubeconfig files %v do not exist on your local machine, thus kubectl authentication won't work in the container", inexistentKubeconfigFiles)
+			}
+		} else {
+			// We only want to use the default kube context if KUBECONFIG env var is not explicitly set.
+			// Its default location is `~/.kube`, see https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/
+			defaultKubeconfigPath := path.Join(os.Getenv("HOME"), ".kube")
+			if _, err := os.Stat(defaultKubeconfigPath); !os.IsNotExist(err) {
+				localArgs = append(localArgs, "-v", defaultKubeconfigPath+":"+kubectlDefaultConfigMountPath)
+				localArgs = append(localArgs, "-e", kubeconfigEnvKey+"="+kubectlDefaultConfigMountPath+string(os.PathSeparator)+"config")
+			} else {
+				logrus.Warning("The default kube context does not exist on your local machine, thus gloud authentication won't work in the container")
+			}
+		}
+	}
+
+	for _, env := range container.Env {
+		// At this point, the best effort would have been made to setup gcloud and kubectl credentials if requested,
+		// so skip the env var here to avoid overwriting.
+		if env.Name == gcloudCredEnvName || env.Name == kubeconfigEnvKey {
+			continue
+		}
+		if strExists(opt.skippedEnvVars, env.Name) {
+			continue
+		}
+		localArgs = append(localArgs, "-e", env.Name+"="+env.Value)
+	}
+	for envKey, envValue := range opt.extraEnvVars {
+		localArgs = append(localArgs, "-e", envKey+"="+envValue)
```

Comment:
What would happen if this is meant to be an override? would `-e foo=bar -e foo=baz` work?

Reply:
I have changed to use a map to keep track of all the envs, and overwrite whenever needed, as we chatted offline. So there won't be duplication.